From b68ff1f53dc2ccb9849b6ea01b3b53ad16bd621a Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Wed, 28 Aug 2013 12:09:54 +0800
Subject: [PATCH 04/17] normalize rules with RFC 5234

Drop LF, SP which are defined in RFC 5234.

Replace HT with HTAB (also defined in the RFC).

Use '/' instead of '|', as the RFC does.
---
 Documentation/technical/http-protocol.txt | 26 +++++++++-----------------
 1 file changed, 9 insertions(+), 17 deletions(-)

diff --git a/Documentation/technical/http-protocol.txt b/Documentation/technical/http-protocol.txt
index 55753bb..ff91bb0 100644
--- a/Documentation/technical/http-protocol.txt
+++ b/Documentation/technical/http-protocol.txt
@@ -192,16 +192,13 @@ according to the C locale ordering.  The file SHOULD NOT include
 the default ref named 'HEAD'.
 
   info_refs        =  *( ref_record )
-  ref_record       =  any_ref | peeled_ref
+  ref_record       =  any_ref / peeled_ref
 
-  any_ref          =  id HT name LF
-  peeled_ref       =  id HT name LF
-		      id HT name "^{}" LF
+  any_ref          =  id HTAB name LF
+  peeled_ref       =  id HTAB name LF
+		      id HTAB name "^{}" LF
   id               =  40*HEX
 
-  HEX              =  "0".."9" | "a".."f"
-  LF               =  <US-ASCII LF, linefeed (10)>
-  HT               =  <US-ASCII HT, horizontal-tab (9)>
 
 Smart Clients
 ~~~~~~~~~~~~~
@@ -288,7 +285,7 @@ declarations behind a NUL on the first ref.
   smart_reply      =  PKT-LINE("# service=$servicename" LF)
 		      ref_list
 		      "0000"
-  ref_list         =  empty_list | non_empty_list
+  ref_list         =  empty_list / non_empty_list
 
   empty_list       =  PKT-LINE(id SP "capabilities^{}" NUL cap_list LF)
 
@@ -296,18 +293,13 @@ declarations behind a NUL on the first ref.
 		      *ref_record
 
   cap_list         =  *(SP capability) SP
-  ref_record       =  any_ref | peeled_ref
+  ref_record       =  any_ref / peeled_ref
 
   any_ref          =  PKT-LINE(id SP name LF)
   peeled_ref       =  PKT-LINE(id SP name LF)
 		      PKT-LINE(id SP name "^{}" LF
   id               =  40*HEX
 
-  HEX              =  "0".."9" | "a".."f"
-  NL               =  <US-ASCII NUL, null (0)>
-  LF               =  <US-ASCII LF,  linefeed (10)>
-  SP               =  <US-ASCII SP,  horizontal-tab (9)>
-
 
 Smart Service git-upload-pack
 ------------------------------
@@ -343,7 +335,7 @@ appear in the response obtained through ref discovery.
   compute_request  =  want_list
 		      have_list
 		      request_end
-  request_end      =  "0000" | "done"
+  request_end      =  "0000" / "done"
 
   want_list        =  PKT-LINE(want NUL cap_list LF)
 		      *(want_pkt)
@@ -353,7 +345,7 @@ appear in the response obtained through ref discovery.
 
   have_list        =  *PKT-LINE("have" SP id LF)
 
-  command          =  create | delete | update
+  command          =  create / delete / update
   create           =  40*"0" SP new_id SP name
   delete           =  old_id SP 40*"0" SP name
   update           =  old_id SP new_id SP name
@@ -530,7 +522,7 @@ the id obtained through ref discovery as old_id.
   command_pkt      =  PKT-LINE(command LF)
   cap_list         =  *(SP capability) SP
 
-  command          =  create | delete | update
+  command          =  create / delete / update
   create           =  40*"0" SP new_id SP name
   delete           =  old_id SP 40*"0" SP name
   update           =  old_id SP new_id SP name
-- 
1.8.4.rc4.527.g303b16c

